target/ppc/translate.c

```diff
@@ -7452,25 +7452,36 @@ static int times_4(DisasContext *ctx, int x)
  * Use token pasting to ensure that we use the proper flag with the
  * proper variable.
  */
-#define REQUIRE_INSNS_FLAGS(CTX, NAME) \
-    do {                                                \
-        if (((CTX)->insns_flags & PPC_##NAME) == 0) {   \
-            return false;                               \
-        }                                               \
+#define REQUIRE_INSNS_FLAGS_INTERNAL(CTX, FLAGS)         \
+    do {                                                 \
+        if (((CTX)->insns_flags & (FLAGS)) != (FLAGS)) { \
+            return false;                                \
+        }                                                \
     } while (0)
-
-#define REQUIRE_INSNS_FLAGS2(CTX, NAME) \
-    do {                                                \
-        if (((CTX)->insns_flags2 & PPC2_##NAME) == 0) { \
-            return false;                               \
-        }                                               \
+#define REQUIRE_INSNS_FLAGS2_INTERNAL(CTX, FLAGS2)          \
+    do {                                                    \
+        if (((CTX)->insns_flags2 & (FLAGS2)) != (FLAGS2)) { \
+            return false;                                   \
+        }                                                   \
     } while (0)
+#define REQUIRE_INSNS_FLAGS(CTX, F1) \
+    REQUIRE_INSNS_FLAGS_INTERNAL((CTX), PPC_##F1)
+#define REQUIRE_INSNS_FLAGS_2(CTX, F1, F2) \
+    REQUIRE_INSNS_FLAGS_INTERNAL((CTX), PPC_##F1 | PPC_##F2)
```

Comment:
I'm not sure this is is the intended use of `insns_flags`/`insns_flags2`. Do any instructions check more than one bit in each flag?

Reply:
Good point. I thought that case existed, but couldn't find any. Fixup in https://github.com/PPC64/qemu/pull/31/commits/eb21acc6b7e5afa26d61195b972e039077674e0d